site/cypress/e2e/Cypress-UI/navigation_page.spec.js

```diff
@@ -9,6 +7,91 @@ function validate_navigation_page_sections(sections) {
+const red = 'rgb(220, 53, 69)';
+const green = 'rgb(95, 183, 96)';
+const buttonOrder = ['team-btn', 'submit-btn', 'grade-btn', 'quick-link-btn'];
+
+function checkButtons(courseID, buttonText, buttonColor) {
+    // assert that the course does not exist
+    if (buttonText === null && buttonColor === null) {
+        cy.get(`[data-testid="${courseID}"]`).should('not.exist');
+    }
+    else {
+        for (let i = 0; i < buttonOrder.length; i++) {
+            if (buttonText[i]) {
+                cy.get(`[data-testid="${courseID}"]`).find(`[data-testid="${buttonOrder[i]}"]`).contains(buttonText[i]);
+                cy.get(`[data-testid="${courseID}"]`).find(`[data-testid="${buttonOrder[i]}"]`).should('have.css', 'background-color', buttonColor[i]);
+            }
+            else {
+                // assert that the button does not exist
+                cy.get(`[data-testid="${courseID}"]`).find(`[data-testid="${buttonOrder[i]}"]`).should('not.exist');
+            }
+        }
+    }
+}
+
+describe('tests navigation buttons for each level of access', () => {
+    beforeEach(() => {
+        cy.visit('/');
+        cy.viewport(1920, 1200);
+    });
+    // Future (No TAs) Homework, Future(TAs) Homework, Open Homework, Open Team Homework, Closed Homework, Closed Team Homework
+    // grading homework, Grading Lab
+    // Grade Released Homework, Grade Released Test,
+    it('should display instructor buttons correctly', () => {
+        cy.login('instructor');
+        cy.visit(['sample']);
+
+        checkButtons('future_no_tas_homework', [null, 'ALPHA SUBMIT', 'PREVIEW GRADING', 'OPEN TO TAS NOW'], [null, white, white, blue]);
+        checkButtons('future_tas_homework', [null, 'BETA SUBMIT', 'PREVIEW GRADING', 'OPEN NOW'], [null, white, white, blue]);
+        checkButtons('open_homework', [null, 'SUBMIT', 'PREVIEW GRADING', 'CLOSE SUBMISSIONS NOW'], [null, blue, white, white]);
+        checkButtons('open_team_homework', ['CREATE TEAM', 'MUST BE ON A TEAM TO SUBMIT', 'PREVIEW GRADING', 'CLOSE SUBMISSIONS NOW'], [blue, blue, white, white]);
+        checkButtons('closed_homework', [null, 'LATE SUBMIT', 'PREVIEW GRADING', 'OPEN TO GRADING NOW'], [null, red, white, blue]);
+        checkButtons('closed_team_homework', ['CREATE TEAM', 'MUST BE ON A TEAM TO SUBMIT', 'PREVIEW GRADING', 'OPEN TO GRADING NOW'], [red, red, white, blue]);
+        checkButtons('grading_homework', [null, 'OVERDUE SUBMISSION', 'GRADE', 'RELEASE GRADES NOW'], [null, red, blue, blue]);
+        checkButtons('grading_lab', [null, null, 'GRADE', 'RELEASE GRADES NOW'], [null, null, white, blue]);
+        checkButtons('grades_released_homework_autota', [null, 'VIEW GRADE', 'REGRADE', null], [null, green, white, null]);
+        checkButtons('grades_released_homework', [null, 'OVERDUE SUBMISSION', 'REGRADE', null], [null, red, white, null]);
+        checkButtons('grades_released_lab', [null, null, 'REGRADE', null], [null, null, white, null]);
```

Comment:
It might be easier to read if these were split up and have each parameter on differentlines, or have whitespace between each line but I'm not hard on that, it is just a bit cluttered.